editor/src/messages/tool/tool_messages/path_tool.rs

```diff
@@ -285,11 +343,34 @@ impl Fsm for PathToolFsmState {
 				}
 
 				// Mouse up
-				(_, PathToolMessage::DragStop { shift_mirror_distance }) => {
+				(state, PathToolMessage::DragStop { shift_mirror_distance } | PathToolMessage::Enter { shift_mirror_distance }) => {
+					let shift_pressed = input.keyboard.get(shift_mirror_distance as usize);
+
+					if state == PathToolFsmState::DrawingBox {
+						let mut finished_processing = false;
+						if tool_data.drag_start == tool_data.drag_current {
+							responses.add(DocumentMessage::DeselectAllLayers);
+							finished_processing = true;
+						}
+
+						if !finished_processing {
+							let quad = tool_data.selection_quad();
+							shape_editor.select_all_in_quad(&document.document_legacy, quad.bounding_box(), !shift_pressed);
```

Comment:
We don't need to create a quad, we could just pass in `[self.drag_start, self.drag_current]`.